postgres/datadog_checks/postgres/statements.py

```diff
@@ -0,0 +1,181 @@
+            return
+
+        desired_columns = (
+            list(PG_STAT_STATEMENTS_METRIC_COLUMNS.keys())
+            + list(PG_STAT_STATEMENTS_OPTIONAL_COLUMNS)
+            + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        query_columns = list(
+            set(desired_columns) & set(available_columns) + list(PG_STAT_STATEMENTS_TAG_COLUMNS.keys())
+        )
+        rows = self._execute_query(
+            db.cursor(cursor_factory=psycopg2.extras.DictCursor),
+            STATEMENTS_QUERY.format(
+                cols=', '.join(query_columns),
+                pg_stat_statements_view=self.config.pg_stat_statements_view,
+            ),
+            params=(self.config.dbname,),
+        )
+        if not rows:
+            return
+
+        def row_keyfunc(row):
+            # old versions of pg_stat_statements don't have a query ID so fall back to the query string itself
+            queryid = row['queryid'] if 'queryid' in row else row['query']
+            return (queryid, row['datname'], row['rolname'])
+
+        rows = self._state.compute_derivative_rows(rows, PG_STAT_STATEMENTS_METRIC_COLUMNS.keys(), key=row_keyfunc)
+        metric_limits = (
+            self.config.statement_metric_limits if self.config.statement_metric_limits else DEFAULT_METRIC_LIMITS
+        )
+        rows = apply_row_limits(rows, metric_limits, 'calls', True, key=row_keyfunc)
+
+        for row in rows:
+            try:
+                normalized_query = datadog_agent.obfuscate_sql(row['query'])
+                if not normalized_query:
+                    logger.warning("Query obfuscation resulted in empty query '%s'", row['query'])
+                    continue
+            except Exception as e:
+                logger.warning("Failed to obfuscate query '%s': %s", row['query'], e)
+                continue
+
+            # The APM resource hash will use the same query signature because the grouped query is close
+            # enough to the raw query that they will intersect frequently.
+            query_signature = compute_sql_signature(normalized_query)
+            tags = ['query_signature:' + query_signature, 'resource_hash:' + query_signature] + instance_tags
+            for column, tag_name in PG_STAT_STATEMENTS_TAG_COLUMNS.items():
+                if column not in row:
+                    continue
+                value = row[column]
+                if column == 'query':
+                    value = self._normalize_query_tag(value)
+                tags.append('{tag_name}:{value}'.format(tag_name=tag_name, value=value))
+
+            for column, metric_name in PG_STAT_STATEMENTS_METRIC_COLUMNS.items():
+                if column not in row:
+                    continue
+                value = row[column]
+                if column == 'total_time':
+                    # convert milliseconds to nanoseconds
+                    value = value * 1000000
+                instance.count(metric_name, value, tags=tags)
```

Comment:
I believe the metrics should be sent inside the `PostgreSql` class. This method should return the list of metrics we want to send instead of sending them directly (see `PostgresMetricsCache`).
Then, we can submit them in `PostgresSql. _collect_per_statement_metrics` and avoid passing `instance` as an argument to `PostgresStatementMetrics. collect_per_statement_metrics`.